### PR TITLE
Update all dependencies in rust bitcoin ecosystem

### PR DESCRIPTION
**Pull request by sanket1729**
Hi @stevenroose, 
   I was not aware of this project until Andrew mentioned it to be last week. Hal is very cool. I have updated all the dependencies to the latest releases. 

There were a lot of changes in rust-miniscript, there is a big diff. Another commit adds more features from rust-miniscript. Over the next weekend, I will add support for xpub descriptors. 

I also want to add a command to the psbt module something like:

`signrawpsbt <psbt> <index> <list_priv_keys>` that would signed psbt.

### CODE REVIEW

**Comment by stevenroose** [comment]
Thanks! I've been having a hard time contemplating useful miniscript commands because I didn't really know the APIs (they also seem to have changed a lot). So very happy to have you add some!

Can you remove the Cargo.lock commit? Since this is a binary project, it's supposed to be included (please also commit changes to it in the commit that updates the dependencies), so that the build doesn't break when dependencies change.

File: src/bin/hal/cmd/miniscript.rs [hunk 1]
```diff
@@ -279,6 +275,12 @@ impl FromScriptContexts for MiniscriptInfo {
 					None
 				}
 			},
+			requires_sig: ms.requires_sig(),
+		    non_malleable: ScriptContexts::from_bare(ms.is_non_malleable()),
```
Comment (stevenroose) [inline src/bin/hal/cmd/miniscript.rs:279]: some indentation here

File: src/bin/hal/cmd/miniscript.rs [hunk 2]
```diff
@@ -306,6 +304,12 @@ impl FromScriptContexts for MiniscriptInfo {
 					None
 				}
 			},
+			requires_sig: ms.requires_sig(),
+		    non_malleable: ScriptContexts::from_p2sh(ms.is_non_malleable()),
```
Comment (stevenroose) [inline src/bin/hal/cmd/miniscript.rs:308]: indentation

File: src/bin/hal/cmd/miniscript.rs [hunk 3]
```diff
@@ -333,6 +333,12 @@ impl FromScriptContexts for MiniscriptInfo {
 					None
 				}
 			},
+			requires_sig: ms.requires_sig(),
+		    non_malleable: ScriptContexts::from_segwitv0(ms.is_non_malleable()),
```
Comment (stevenroose) [inline src/bin/hal/cmd/miniscript.rs:337]: indentation

File: Cargo.toml [hunk 4]
```diff
@@ -31,11 +31,11 @@ serde_json = "1.0.34"
...
 hex = "0.3.2"
 
-bitcoin = { version = "0.23", features = [ "use-serde" ] }
-secp256k1 = { version = "0.17", features = [ "recovery" ] }
-tiny-bip39 = "0.6.2"
-lightning-invoice = "0.2.0"
-miniscript = { version = "0.12.0", features = ["compiler"] }
+bitcoin = { version = "0.26.0", features = [ "use-serde" ] }
+secp256k1 = { version = "0.20.1", features = [ "recovery" ] }
+tiny-bip39 = "0.8.0"
```
Comment (stevenroose) [inline Cargo.toml:36]: I'm planning to use the bip39 crate here soon (as soon as v1.0.0 releases), so feel free to revert this.

**Review by stevenroose — CHANGES_REQUESTED** [review]
Other than Cargo.lock and some nits, this looks good! Thanks!

**Comment by sanket1729** [comment]
Ready for review again

**Review by stevenroose — APPROVED** [review]
LGTM, thanks!

**Comment by stevenroose** [comment]
> I also want to add a command to the psbt module something like:
> 
> signrawpsbt <psbt> <index> <list_priv_keys> that would signed psbt.

About that, yeah that and finalizepsbt were 2 commands I really wanted but needed miniscript features to actually fulfill. Would be awesome if you could add them once miniscript supports them!

**Pull request merged**